### PR TITLE
LF: Drop old deprecate definitions

### PR DESCRIPTION
**Pull request by remyhaemmerle-da**
CHANGELOG_BEGIN
CHANGELOG_END

### CODE REVIEW

File: daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/ContractStateMachine.scala [hunk 1]
```diff
@@ -242,7 +242,7 @@ class ContractStateMachine[Nid](mode: ContractKeyUniquenessMode) {
       }
       Either.cond(
         keyMapping == keyResolution,
-        next,
+        21 next,
```
Comment (remyhaemmerle-da) [inline daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/ContractStateMachine.scala:245]:
```suggestion
        next,
```

File: daml-lf/transaction/src/test/scala/com/digitalasset/daml/lf/value/ValueCoderSpec.scala [hunk 2]
```diff
@@ -155,7 +155,7 @@ class ValueCoderSpec
     }
 
     "do versioned value with supported override version" in forAll(versionedValueGen) {
-      case VersionedValue(version, value) => testRoundTrip(value, version)
+      case Versioned(version, value) => testRoundTrip(value, version)
```
Comment (remyhaemmerle-da) [inline daml-lf/transaction/src/test/scala/com/digitalasset/daml/lf/value/ValueCoderSpec.scala:158]: @adriaanm-da Another deprecate usage not caught by the compiler.

**Review by garyverhaegen-da — APPROVED** [review]

**Pull request merged**